src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -199,6 +199,30 @@ public Assembly LoadFromAssemblyName(AssemblyName assemblyName)
             return assembly;
 
         }
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        [SuppressUnmanagedCodeSecurity]
+        private static extern IntPtr LoadUnmanagedLibraryFromPath(string unmanagedDllPath);
+
+        // This method provides a way for overriders of LoadUnmanagedDll() to load an unmanaged DLL from a specific path in a
+        // platform-independent way. The DLL is loaded with default load flags.
+        protected IntPtr LoadUnmanagedDllFromPath(string unmanagedDllPath)
+        {
+            if (unmanagedDllPath == null)
+            {
+                throw new ArgumentNullException("unmanagedDllPath");
+            }
+            if (unmanagedDllPath.Length == 0)
+            {
+                throw new ArgumentException(Environment.GetResourceString("Argument_EmptyPath"), "unmanagedDllPath");
+            }
+            if (Path.IsRelative(unmanagedDllPath))
```

Comment:
I am not sure whether we should be in the business of enforcing the full path here. What is the harm in letting the relative paths through - and letting the default platform probing policy kick in for them?